### PR TITLE
Support loading check.env

### LINKED ISSUE #12: Whitelist support?
Checking of package isn't new or limited to R.  Debian has been doing it since time immortal as well with a tool called 'lintian'.  One _really_ useful feature is a whitelist one can add per-package.  Its format is really simple:  one item per row, and the item is the warning message.   So before counting / printing the warning, the override file is checked.

I think we should brainstorm how to support that. Maybe a file `inst/rccmdcheck.overrides.txt` ?  And in the output structure then have whitelist-adjusted counts?

### PR DESCRIPTION
**Pull request by gaborcsardi**
Will add docs...

Closes #12.

### CODE REVIEW

File: R/env.R [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    )
+    env_path <- file.path(tmp, package, "inst", "check.env")
+  }
+
+  if (file.exists(env_path)) {
+    load_env_file(env_path, envir = envir)
+  }
+}
+
+load_env_file <- function(path, envir = parent.frame()) {
```
Comment (jimhester) [inline R/env.R:28]: Should this use `readRenviron()` instead of using our own implementation?

**Comment by hadley** [comment]
This doesn't completely capture what I was looking for, which is some way to disable a standard set of notes that we don't think is important. i.e. this looks useful, but doesn't directly solve the problem I care about.

File: R/env.R (hunk 1 above)
Comment (gaborcsardi) [reply R/env.R:28]: > I guess the downside would be restoring the old envvars afterwards. But you could do Sys.getenv() to save them beforehand I guess.

I don't really like its way of handling errors, it just ignores them, and especially on a CI, it is better to know about them.

So it was easier to copy this code from another package of mine. 

But I don't really mind, we can switch to `readRenviron()` as well.

**Comment by gaborcsardi** [comment]
> This doesn't completely capture what I was looking for, which is some way to disable a standard set of notes that we don't think is important. i.e. this looks useful, but doesn't directly solve the problem I care about.

I would wary about _always_ ignoring these notes. E.g. the size note is definitely useful sometimes. With the `check.env` file, you can suppress the useless package size notes by setting a reasonable limit for a package.

But we can have that as well. Do you want to make this opt-in or opt-out? We can have a `Config/rcmdcheck/ignore_notes` flag for it?

**Comment by gaborcsardi** [comment]
@hadley now you can set 
```
Config/rcmdcheck/ignore-inconsequential-notes: true
```
in `DESCRIPTION`. It sets these env vars currently:
```
  c("_R_CHECK_PKG_SIZES_" = "false",
    "_R_CHECK_RD_XREFS_" = "false",
    "_R_CHECK_CRAN_INCOMING_NOTE_GNU_MAKE_" = "false",
    "_R_CHECK_PACKAGE_DATASETS_SUPPRESS_NOTES_" = "false"
  )
```

Is this closer to what you had in mind?

**Comment by hadley** [comment]
Yeah, that's better. We can then change the `check-full` default to error with notes, and then there's a standard way to suppress either the most common unimportant notes, or selected notes affecting your package.

Then this PR just needs a bit of documentation so it's easy to find out these two ways of suppressing notes.

**Comment by gaborcsardi** [comment]
@hadley @jimhester I have added some docs.

I also removed the dependence on `NOT_CRAN`. I think if somebody adds `check.env` _and_ uses rcmdcheck, then they surely want to use it.

**Comment by jimhester** [comment]
I think we might still need to have some sort of non envvar based allow list. Even though CRAN has been adding more fine grained ennvars lately older versions of R won't have this, so we will still get false positive NOTES on those versions. This would preclude us from changing the default to failing on NOTEs for our action

| name | versions |
|------|----------|
| `_R_CHECK_PKG_SIZES_` | 3.6.0+ |
| `_R_CHECK_RD_XREFS_` | 3.0.3+ |
|`_R_CHECK_CRAN_INCOMING_NOTE_GNU_MAKE_` | 3.6.0+ |
| `_R_CHECK_PACKAGE_DATASETS_SUPPRESS_NOTES_` | 3.3.0+ |

We are currently at 3.4 as the last version we are including in the checks, so it will still be a while before we could use this unconditionally.

**Comment by gaborcsardi** [comment]
`_R_CHECK_PKG_SIZES_` is included in R 3.4.x, I think, even in the R 3.0.x branch, actually. The other env var for the threshold is more recent.

As for the one for GNU make, yeah, that is a pity. But if it is the only one, then maybe it is not worth implementing a(nother) ignore mechanism for this single one.

We do use GNU make in a number of repos, so in these we could have `error_on = warning` on the older R versions. (I'll add support to set `error_on` from an env var.)

**Comment by jimhester** [comment]
Ah, yes you are right about `_R_CHECK_PKG_SIZES_`, it is `_R_CHECK_PKG_SIZES_THRESHOLD_` that is only on 3.6+

**Comment by gaborcsardi** [comment]
@jimhester So do your think that it would be OK to add support for the `error_on = "warning"` workaround on older R versions? Then we don't need a generic ignore-list feature.

Unfortunately it is hard to do an ignore-list properly.

**Comment by jimhester** [comment]
yeah I think that is the best solution for now, we can maybe just fall back to using `error_on` warning for packages that use GNU Makefiles.

**Pull request merged**